polygeist/lib/Dialect/Polygeist/Transforms/ArgumentPromotion.cpp

```diff
@@ -91,45 +91,6 @@ static bool isValidMemRefType(Type type) {
   return true;
 }
 
-// Returns true if the block of \p startOp contains an operation (after startOp)
```

Comment:
no longer needed if we limit candidate to be tail call. I can add it back if preferred. 
But then we cannot check for tail call on the new utility function to determine if the given function is a potential kernel body function.

Reply:
I do not understand why it was removed, I designed the pass to allow unaliased load/stores after the tail call to be more generic than necessary currently as to "future proof" this transformation. Can you please add it back.